### DISCUSSION TITLE
PHP 8.1 ExceptionError

### DISCUSSION THREAD

**Discussion by onomica** [voku/anti-xss#123] [General]
steps: 
1. $str = '%90' || $str = '8D' || $str = '9D'
2. executed public function xss_clean($str)
actual result :
 - Deprecated: Implicit conversion from float 2.25 to int loses precision

problem here: voku/portable-utf8/src/voku/helper/UTF8.php:13888

**Comment by voku** [comment]
I think the idea was to decode everything, so that we can identify XSS attacks, if you didn't find anything, we can return the original value. 🤔 Can you some failing tests, so that we can work on that, thanks.

**Reply by onomica** [reply]
i get error every time if string existed this text ('%90' '%8D' '%9D')
in example string '%90'
voku/portable-utf8/src/voku/helper/UTF8.php
in this file line 13884

$ordC1 = self::$ORD[$input];
$input - this symbol fron ascii table
$ordC1 - code ascii symobl  (for %90 this code is 144)

$cc1 = self::$CHR[$ordC1 / 64] | "\xC0";
here we get error, because 144/64 = 2.25

self::$CHR[2.25] - float key and here get error "Deprecated: Implicit conversion from float 2.25 to int loses precision"

upd: self::$CHR contains all ascii symbols in this format 'number' => 'symbol'
$ordC1 - number of ascii table 
why need divide by 64 ?

**Reply by onomica** [reply]
any update?
